app/forms/user_piv_cac_setup_form.rb

```diff
@@ -0,0 +1,80 @@
+class UserPivCacSetupForm
+  # extend Forwardable
+  include ActiveModel::Model
+
+  attr_accessor :x509_dn_uuid, :x509_dn, :token, :user, :nonce, :error_type
+
+  validates :token, presence: true
+  validates :nonce, presence: true
+  validates :user, presence: true
+
+  def submit
+    success = valid? && valid_token?
+
+    FormResponse.new(success: success && process_valid_submission, errors: {})
+  end
+
+  private
+
+  def process_valid_submission
+    user.x509_dn_uuid = x509_dn_uuid
+    user.save!
+    Event.create(user_id: user.id, event_type: :piv_cac_enabled)
+    true
+  rescue PG::UniqueViolation
+    self.error_type = 'piv_cac.already_associated'
+    false
+  end
+
+  def valid_token?
+    user_has_no_piv_cac &&
+      token_decoded &&
+      token_has_correct_nonce &&
+      not_error_token &&
+      piv_cac_not_already_associated
+  end
```

Comment:
Anything that isn't broken I'd like to take on as technical debt for later this sprint so we can get something out for move.mil.

The conversion would require reworking the translations and the error pages since we pull in translated text based on the error.

Reply:
OK, sounds good.

Reply:
But wait, we already packaged the release. Are we planning to do a patch release?

Reply:
Yep